LCAPI.Core/Patches/GameNetworkManager.cs

```diff
@@ -1,54 +1,83 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-using Mono.Cecil;
-using BepInEx;
-using UnityEngine;
+﻿namespace LCAPI.Core.Patches;
+
+using HarmonyLib;
 using Steamworks;
 using Steamworks.Data;
-using HarmonyLib;
-using HarmonyLib.Tools;
+using System.Linq;
 
-namespace LCAPI.Core.Patches
+[HarmonyPatch(typeof(GameNetworkManager))]
+internal class GameNetworkManagerPatches
 {
-    [HarmonyPatch(typeof(GameNetworkManager))]
-    internal class GameNetworkManagerPatches
+
+    [HarmonyPatch("SteamMatchmaking_OnLobbyCreated")]
+    [HarmonyPostfix]
+    [HarmonyWrapSafe]
+    private static void SteamMatchmaking_OnLobbyCreated_Postfix(Result result, ref Lobby lobby)
     {
-        [HarmonyPatch("SteamMatchmaking_OnLobbyCreated")]
-        [HarmonyPrefix]
-        [HarmonyWrapSafe]
-        private static bool SteamMatchmaking_OnLobbyCreated_Prefix(GameNetworkManager __instance, Result result, Lobby lobby)
+        // lobby has not yet created or something went wrong
+        if (result != Result.OK)
         {
-            if (result != Result.OK)
-            {
-                Debug.LogError(string.Format("Couldn't create lobby {0}!", result), __instance);
-                Plugin.Log.LogError(string.Format("Couldn't create lobby {0}!", result));
-                return false;
-            }
+            return;
+        }
+
+        lobby.SetData("__modded_user", "true");
```

Comment:
Could you run me through your reasoning / ideas behind this system?

Reply:
To explain, normal user can't join a lobby that have metadata `joinable` set to false, so i 'abuse' this trick to make normal player can't join the modded lobby.

The modded user have additional check and it use `__joinable` instead of the `joinable` one, they also can't join lobby that doesn't have the `__modded_user` metadata